### ISSUE TITLE
Implement debug_traceCall

### ISSUE THREAD

**Issue by Wodann** [NomicFoundation/edr#192]
Building on the work done in #65, add support for the `debug_traceCall` API by using the same tracer to operate on a single transaction

**Definition of Done**
- Implement the `debug_traceCall` functionality in `rethnet_evm`
- Expose the `trace_call` function through the NAPI and use it in the stubbed version of the EDR VM adapter

**Comment by agostbiro** [comment]
Questions:

- `debug_traceCall` is implemented in `main`, but not in `rethnet/main`. Mergin `debug_traceCall` to `rethnet/main` is a blocker.
- Assuming we want to test against the [fixtures](https://github.com/NomicFoundation/hardhat/tree/781161c00a467a1da392d684950eec4d8574b868/packages/hardhat-core/test/fixture-debug-traces/traceCall) in `main`. It's unclear why some of these have `stack`, `memory` and `storage` fields and others don't. 
- I'm assuming we want EIP-3155 compatible output in EDR for this feature?
- I noticed that there are a lot of adjustments made in [`_runTxAndRevertMutations`](https://github.com/NomicFoundation/hardhat/blob/781161c00a467a1da392d684950eec4d8574b868/packages/hardhat-core/src/internal/hardhat-network/provider/node.ts#L2464) in `main` that were moved into the EthereumJS VM in `rethnet/main`. It's possible that there will be unexpected issues due to this when exposing `debug_traceCall` support in EDR.
- The way `debug_traceTransaction` is implemented currently is not ideal and doesn't have support for gas cost of calls and creates. The problem is that the current implementation just adopts the REVM `TracerEIp3155` implementation that operates conceptually on the opcode level. It already requires some hackery to make this approach work for calls/creates, and it breaks down for call gas cost calculation. A better approach would be to start a new trace for each call/create and then flatten the results to the format that EIP-3155 prescribes. To achieve this, I’d do a refactoring of `rethnet_evm::trace` & `rethnet_evm::debug_trace` to have a single trace collector that collects all the info needed for trace and debug_trace and then have methods that output it in the desired formats. Do we want to rectify that as part of this PR?

**Comment by Wodann** [comment]
1. The merge is almost done. It's in the final stage of review
2. It might be related to the "disable*" booleans. If it doesn't comply with the spec, Franco Victorio is happy to fix the Hardhat side
3. Yes, that'd be great.
4. This should have been resolved in the aforementioned merge that Franco & I did
5. As we need `debug_trace` for the release, I'd opt for the shortest path to completion and do the refactor afterwards
